src/wfs/src/test/java/org/geoserver/wfs/response/CSVOutputFormatTest.java

```diff
@@ -68,6 +70,28 @@
 public class CSVOutputFormatTest extends WFSTestSupport {
     private static final String CSV = "text/csv";
 
+    @Test
+    public void testWithAttributesRemoved() throws Exception {
+        String layerId = getLayerId(ROAD_SEGMENTS);
+        FeatureTypeInfo fti = getCatalog().getFeatureTypeByName(layerId);
+        fti.getAttributes().addAll(fti.attributes());
+        fti.getAttributes().remove(1);
+        getCatalog().save(fti);
+
+        GeoServer gs = getGeoServer();
+        WFSInfo wfs = gs.getService(WFSInfo.class);
+        wfs.setFeatureBounding(false);
+        gs.save(wfs);
+
+        MockHttpServletResponse response =
+                getAsServletResponse(
+                        "ows?service=WFS&version=1.0.0&request=GetFeature&typeName=cite:RoadSegments&maxFeatures=50&outputFormat=text%2Fcsv&propertyname=NAME");
+        assertEquals(
+                "attachment; filename=RoadSegments.csv", response.getHeader("Content-Disposition"));
+
+        assertEquals(CSV, getBaseMimeType(response.getContentType()));
+    }
```

Comment:
this does not actually test if the output only has the NAME column, it only tests if a csv file is produced